server/api.js

```diff
@@ -551,6 +616,48 @@ router.postAsync("/match", ensure.isAdmin, async (req, res) => {
   res.send(match);
 });
 
+/**
+ * POST /api/warmup
+ * Submit a warmup
+ * Params:
+ *   - match: match ID
+ *   - playerNo: 1 - player 1, 2 - player 2
+ *   - warmup: the warmup map, could be beatmap link or ID
+ */
+router.postAsync("/warmup", async (req, res) => {
+  const match = await Match.findOne({ _id: req.body.match });
+  if (!(await canEditWarmup(req.user, req.body.playerNo, match))) {
+    logger.warn(`${req.user.username} tried to submit player ${req.body.playerNo} warmup for ${req.body.match}`);
+    return res.status(403).send("You don't have permission to do that");
+  }
+  try {
+    match[`warmup${req.body.playerNo}`] = await parseWarmup(req.body.warmup);
+  } catch (e) {
+    res.status(400).send(e.message);
```

Comment:
o also you should return here
otherwise it'll keep going and will try to res.send a second time 

and can remove the `await` on the res.send below

Reply:
Oops missed it, committed